src/tequila/overlap/overlap.py

```diff
@@ -0,0 +1,88 @@
+from tequila.circuit.circuit import QCircuit, assign_ctrl
```

Comment:
This looks great. Would suggest to move it somewhere else with the package and unify it a bit with tq.Objective, but I would propose to discuss that in the next call (that's easier and I can get your input on it).